app/controllers/spree/api/products_controller_decorator.rb

```diff
@@ -66,6 +71,11 @@ def product_scope
     scope.includes(:master)
   end
 
+  def import_date_scope
+    import_date = params[:import_date].to_datetime
+    Spree::Variant.where(import_date: import_date..import_date + 24.hours)
+  end
```

Comment:
I don't like making the models big but I think this would be better in the variant model as a AR scope. Definitely, nothing to couple the controller with.

Reply:
Moved to a new scope and tests added :+1: